### PR TITLE
Fix trampoline assembly for build on clang 18 on apple silicon (#54634)

### PR DESCRIPTION
**Pull request by Drvi**
This avoids a: `error: non-private labels cannot appear between .cfi_startproc / .cfi_endproc pairs` error.
That error was introduced in https://reviews.llvm.org/D155245#4657075 see also https://github.com/llvm/llvm-project/issues/72802

(cherry picked from commit a4e793ee31612625d9c48ebd89a53c3a5c0f6eb6) (cherry picked from commit 3f35094d298b41d57f3f3a8c5a9abd52bcf47b30)

## PR Description

_What does this PR do?_

## Checklist

Requirements for merging:
- [x] I have opened an issue or PR upstream on JuliaLang/julia: https://github.com/JuliaLang/julia/pull/54634
- [x] I have removed the `port-to-*` labels that don't apply.
- [x] I have opened a PR on raicode to test these changes: https://github.com/RelationalAI/raicode/pull/22029

### CODE REVIEW

**Review by rbvermaa — APPROVED** [review]

**Comment by d-netto** [comment]
Could you provide some higher-level context on why we are backporting this?

I.e. does this fixe any bug related to building raicode with PackageCompiler on Apple silicon? (That's what I imagine...)

**Comment by Drvi** [comment]
@d-netto Sorry, here is a slack thread for context: https://relationalai.slack.com/archives/C04J1BRVD40/p1731383171631329

TL;DR: with newer clang that came with the recent Sonoma upgrade, we couldn't get Julia 1.10.2 or 1.10.2+RAI to build on Apple silicon (they fixed it in Julia 1.10.5)

**Review by d-netto — APPROVED** [review]
SGTM.

Please open a raicode PR and check the other boxes before merging.
